vault/resource_database_secret_backend_role.go

```diff
@@ -20,121 +24,176 @@ var (
+	consts.FieldKeyBits,
+	consts.FieldSignatureBits,
+	consts.FieldCreationStatements,
+	consts.FieldRevocationStatements,
+	consts.FieldRollbackStatements,
+	consts.FieldRenewStatements,
+}
+
 func databaseSecretBackendRoleResource() *schema.Resource {
 	return &schema.Resource{
-		Create: databaseSecretBackendRoleWrite,
-		Read:   provider.ReadWrapper(databaseSecretBackendRoleRead),
-		Update: databaseSecretBackendRoleWrite,
-		Delete: databaseSecretBackendRoleDelete,
-		Exists: databaseSecretBackendRoleExists,
+		CreateContext: databaseSecretBackendRoleWrite,
+		ReadContext:   provider.ReadContextWrapper(databaseSecretBackendRoleRead),
+		UpdateContext: databaseSecretBackendRoleWrite,
+		DeleteContext: databaseSecretBackendRoleDelete,
 		Importer: &schema.ResourceImporter{
 			State: schema.ImportStatePassthrough,
 		},
 
 		Schema: map[string]*schema.Schema{
-			"name": {
+			consts.FieldName: {
 				Type:        schema.TypeString,
 				Required:    true,
 				ForceNew:    true,
 				Description: "Unique name for the role.",
 			},
-			"backend": {
+			consts.FieldBackend: {
 				Type:        schema.TypeString,
 				Required:    true,
 				ForceNew:    true,
 				Description: "The path of the Database Secret Backend the role belongs to.",
 			},
-			"db_name": {
+			consts.FieldDBName: {
 				Type:        schema.TypeString,
 				Required:    true,
 				ForceNew:    true,
 				Description: "Database connection to use for this role.",
 			},
-			"default_ttl": {
+			consts.FieldDefaultTTL: {
 				Type:        schema.TypeInt,
 				Optional:    true,
 				Description: "Default TTL for leases associated with this role, in seconds.",
 			},
-			"max_ttl": {
+			consts.FieldMaxTTL: {
 				Type:        schema.TypeInt,
 				Optional:    true,
 				Description: "Maximum TTL for leases associated with this role, in seconds.",
 			},
-			"creation_statements": {
+			consts.FieldCreationStatements: {
 				Type:        schema.TypeList,
 				Required:    true,
 				Elem:        &schema.Schema{Type: schema.TypeString},
 				Description: "Database statements to execute to create and configure a user.",
 			},
-			"revocation_statements": {
+			consts.FieldRevocationStatements: {
 				Type:        schema.TypeList,
 				Optional:    true,
 				Elem:        &schema.Schema{Type: schema.TypeString},
 				Description: "Database statements to execute to revoke a user.",
 			},
-			"rollback_statements": {
+			consts.FieldRollbackStatements: {
 				Type:        schema.TypeList,
 				Optional:    true,
 				Elem:        &schema.Schema{Type: schema.TypeString},
 				Description: "Database statements to execute to rollback a create operation in the event of an error.",
 			},
-			"renew_statements": {
+			consts.FieldRenewStatements: {
 				Type:        schema.TypeList,
 				Optional:    true,
 				Elem:        &schema.Schema{Type: schema.TypeString},
 				Description: "Database statements to execute to renew a user.",
 			},
+			consts.FieldCredentialType: {
```

Comment:
Since this is Optional and Vault returns a default of `password`, I think we should set this field as `Computed: true`. That way, if a user does not specify a `credential_type` in the TF config, TFVP will use the default value returned by Vault.

Reply:
Good call! This was added after Milena and I met and should be fixed now